src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java

```diff
@@ -53,6 +56,22 @@ private static String getParameterWithIndex(ParameterAccessor accessor, int inde
 		// noinspection ConstantConditions
 		if (parameter != null) {
 
+			parameterValue = convert(parameter);
+		}
+
+		return parameterValue;
+
+	}
+
+	private static String convert(Object parameter) {
+		if (Collection.class.isAssignableFrom(parameter.getClass())) {
+			Collection<?> collectionParam = (Collection<?>) parameter;
+			StringBuilder sb = new StringBuilder("[");
+			sb.append(collectionParam.stream().map(o -> "\"" + convert(o) + "\"").collect(Collectors.joining(",")));
+			sb.append("]");
```

Comment:
This works, but what do you think about
```java
			StringBuilder sb = new StringBuilder("[");
			sb.append(collectionParam.stream().map(o -> {
				if (o instanceof String) {
					String s = (String) o;
					return '"' + convert(o) + '"';
				} else {
					return convert(o);
				}
			}).collect(Collectors.joining(",")));
			sb.append(']');
```
Might be better to only wrap Strings in quotes.

Reply:
Yes, that makes sense. I'll add a test and incorporate your change.

Reply:
Fixed in 8b154dfd , @sothawo